modules/cache.js

```diff
@@ -354,7 +354,20 @@ Cache.prototype.onTransactionsSaved = function(transactions, cb) {
 			return cb(err);
 		});
 	} else {
```

Comment:
`transactionsSaved` event is emitted during block saving process with `block.transactions`, doesn't matter if the block contains transactions or not. Then in this function, we are assigning to `delegateTransaction` only if we can find transactions with type `DELEGATE`. So this `else` statement will be not reached if the block contains a transaction of this type. We should rely only on `transactions` to invalidate cache for `transactionCount`.